### PR TITLE
Lamitest

### PR DESCRIPTION
**Pull request by mjeanson**

### CODE REVIEW

**Comment by mjeanson** [comment]
Cherry-pick the last commit.

File: analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiAnalysisStub.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ * Copyright (c) 2016 EfficiOS inc, Michael Jeanson
```
Comment (alexmonthy) [inline analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiAnalysisStub.java]: Should be "EfficiOS Inc."

File: analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiAnalysisStub.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import org.eclipse.tracecompass.internal.provisional.analysis.lami.core.module.LamiChartModel;
+import org.eclipse.tracecompass.tmf.core.analysis.ondemand.OnDemandAnalysisException;
+import org.eclipse.tracecompass.tmf.core.trace.ITmfTrace;
+
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+
+/**
+ * @author Michael Jeanson
+ *
```
Comment (alexmonthy) [inline analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiAnalysisStub.java]: Remove this empty line, auto-Javadoc automatically adds it and it's annoying!

File: analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiAnalysisStub.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.tracecompass.internal.provisional.analysis.lami.core.module.LamiAnalysis;
+import org.eclipse.tracecompass.internal.provisional.analysis.lami.core.module.LamiChartModel;
+import org.eclipse.tracecompass.tmf.core.analysis.ondemand.OnDemandAnalysisException;
+import org.eclipse.tracecompass.tmf.core.trace.ITmfTrace;
+
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+
+/**
```
Comment (alexmonthy) [inline analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiAnalysisStub.java]: You should add a very quick description of the class (like "Extension of {@link LamiAnalysis} used for tests")

File: analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiAnalysisStub.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    @Override
+    protected synchronized void initialize() {
+        checkMetadata();
+    }
+
+    private static String readLamiFile(String filename) {
+        URL url;
+        String fileContent = "";
+        try {
+            url = new URL("platform:/plugin/org.eclipse.tracecompass.analysis.lami.core.tests/testfiles/lami/" + filename);
```
Comment (alexmonthy) [inline analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiAnalysisStub.java]: You could define URL directly here:

    URL url = new ...

doesn't matter much

File: analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiAnalysisStub.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+            try (InputStream inputStream = url.openConnection().getInputStream()) {
+                BufferedReader in = new BufferedReader(new InputStreamReader(inputStream));
+                fileContent = in.lines().collect(Collectors.joining());
+            }
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        } catch (IOException e1) {
+            e1.printStackTrace();
```
Comment (alexmonthy) [inline analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiAnalysisStub.java]: If you handle multiple exceptions the exact same way, you can use the Java 7 mutli-catch syntax:

    } catch (MalformedURLException | IOException e) {
        ...
    }

File: analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiJsonParserTest.java [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+import org.eclipse.tracecompass.tmf.core.timestamp.TmfTimestamp;
+import org.eclipse.tracecompass.tmf.tests.stubs.trace.TmfTraceStub;
+import org.junit.Test;
+
+
+/**
+ * Test cases to verify the JSON parsing of LamiAnalyses.
+ *
+ * @author Michael Jeanson
+ *
```
Comment (alexmonthy) [inline analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiJsonParserTest.java]: remove this line!

File: analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiJsonParserTest.java [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+ */
+public class LamiJsonParserTest {
+
+    private static final String TRACEPATH = "fake/path/to/trace";
+
+    private class LamiTmfTraceStub extends TmfTraceStub {
```
Comment (alexmonthy) [inline analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiJsonParserTest.java]: Not a big deal, but you could define the class as static too (`private static class ...`). That way there is only one "template" of the class, not one per instance. Typically if an inner class can be static it should.

File: analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiJsonParserTest.java [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+            return TRACEPATH;
+        }
+    }
+
+    /**
+     * Test the metadata parsing.
+     */
+    @Test
+    public void testMetadata() {
+        LamiTmfTraceStub trace = new LamiTmfTraceStub();
```
Comment (alexmonthy) [inline analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiJsonParserTest.java]: I see "analysis" now depends on the test. However "trace" is still the same, that one could remain in `@Before`/`@After`, to do the initialization and dipose() in one place.

File: analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/package-info.java [hunk 9]
```diff
@@ -1,11 +0,0 @@
...
- * Copyright (c) 2016 EfficiOS Inc., Alexandre Montplaisir
- *
- * All rights reserved. This program and the accompanying materials
- * are made available under the terms of the Eclipse Public License v1.0
- * which accompanies this distribution, and is available at
- * http://www.eclipse.org/legal/epl-v10.html
- *******************************************************************************/
-
-@org.eclipse.jdt.annotation.NonNullByDefault
-package org.eclipse.tracecompass.analysis.lami.core.tests;
```
Comment (alexmonthy) [inline analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/package-info.java]: I shouldn't have added this, my bad!

File: analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiJsonParserTest.java (hunk 6 above)
Comment (mjeanson) [inline analysis/org.eclipse.tracecompass.analysis.lami.core.tests/src/org/eclipse/tracecompass/analysis/lami/core/tests/LamiJsonParserTest.java]: Not sure what "line" you are refering to.

**Comment by mjeanson** [comment]
Rebased and added a commit fixing the previous comments.

**Comment by alexmonthy** [comment]
Thanks, squashed into one commit and merged manually.

**Pull request closed without merging**